source/Weather.h

```diff
@@ -19,11 +19,24 @@ PARTICULAR PURPOSE.  See the GNU General Public License for more details.
 
 class Hazard;
 class Visual;
+class Weapon;
+
 
 
 // Weather is used to contain an active system hazard, keeping track of the hazard's
 // lifetime, its strength, and if it should cause any damage.
 class Weather {
+public:
+	class ImpactInfo {
```

Comment:
Something to consider; would it make sense to change the ImpactInfo classes from subclasses of Weather and Projectile to interfaces implemented by Weather and Projectile?

The data they provide would still be the same, but it would save some copying and DamageProfile would just be able to take the data by interface.